### ISSUE TITLE
Verbose debug output with message in response of `sendMail`

### ISSUE THREAD

**Issue by niftylettuce** [nodemailer/nodemailer#877]
Can we have an option, perhaps a Boolean flag (that's `false` by default) in order to see the message sent to `sendMail` function?  Perhaps in the response as `response.message`?

I believe we would need to modify something here <https://github.com/nodemailer/nodemailer/blob/954ab0e8019f5849945f6bf5dfa321121839931e/lib/mailer/index.js#L134>

In particular this affects our users ability to see the rendered email in `email-templates` package, see <https://github.com/niftylettuce/email-templates/issues/299>

I could and most likely will just patch it on my end (as such below - at least in the interim until a solution if any is found with `nodemailer`), but I'm just wondering what the better solution is?

```diff
        const res = await this.config.transport.sendMail(message);
        debug('message sent');
+        res.message = message;
        resolve(res);
```

Then my user could access via `res.message` the original message sent over.

**Comment by niftylettuce** [comment]
I don't think we should do `res.message` since that would conflict with `JSONTransport` using that property in the response.  Maybe `res.originalMessage`?   That's how I'm going to patch on my end for now I think.

**Comment by niftylettuce** [comment]
Another concern - addresses get reformatted, e.g. `res.message` (from JSONTransport) vs. what was originally passed `res.originalMessage`:

```sh
res.message { from: { address: 'niftylettuce+from@gmail.com', name: '' },
  to: [ { address: 'niftylettuce+to@gmail.com', name: '' } ],
  cc: [ { address: 'niftylettuce+cc@gmail.com', name: '' } ],
  bcc: [ { address: 'niftylettuce+bcc@gmail.com', name: '' } ],
  attachments: [],
  subject: 'Test email for niftylettuce',
  text: 'Hi niftylettuce,\nThis is just a test.',
  headers: {},
  messageId: '<3f671fd9-be7d-1007-6a11-9c465720f6cc@gmail.com>' }
res.originalMessage { from: 'niftylettuce+from@gmail.com',
  to: 'niftylettuce+to@gmail.com',
  cc: 'niftylettuce+cc@gmail.com',
  bcc: 'niftylettuce+bcc@gmail.com',
  attachments: [],
  subject: 'Test email for niftylettuce',
  text: 'Hi niftylettuce,\nThis is just a test.' }
```

**Comment by niftylettuce** [comment]
I've temporarily patched and released v3.6.1 of email-templates and added documentation for how to inspect in the meanwhile: <https://github.com/niftylettuce/email-templates#inspect-message>

**Comment by andris9** [comment]
The value you pass to nodemailer is not immutable, plugins can change it, addresses are normalized, values can contain streams that are consumed by the end of the sendMail etc. I don't think there is a failproof way of adding the input besides deep-cloning it (and this would still help with streams). I guess the best way to handle it would be by adding a wrapper around Nodemailer.
